### PR TITLE
feat(abigenrc): Pin abigen version

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Pins the version of abigen used with a `.abigenrc` file.

The `.abigenrc` file can be used to update your version of abigen with `pnpm install:abigen`.

A convenience script is added to update the `.abigenrc` file with `pnpm upgrade:abigen`.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6556** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6556" target="_blank">Graphite</a>  👈
    * **PR #6558** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6558" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6556?utm_source=stack-comment).

File: .abigenrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v1.10.25
```
Comment (clabby) [inline .abigenrc:1]: Can we also enforce that CI uses this version of abigen?

Comment (refcell) [reply .abigenrc:1]: Yes! Wanted to solicit first thoughts before wiring it in, but I think it makes sense - will stack ontop

Comment (refcell) [reply .abigenrc:1]: Done sir 🖖

**Review by mslipper — APPROVED** [review]

**Pull request merged**